docs/feature_flags.md

```diff
@@ -86,3 +86,13 @@ issues upstream.
 
 In future releases, this new service discovery manager will become the default and
 this feature flag will be ignored.
+
+## Prometheus agent
```

Comment:
nit: should we refer to this as Prometheus Agent or Prometheus agent? I think we've been doing both, but it would be good to be consistent.

Reply:
I feel like this is fine as is. I don't like capitalizing everything. It could even be Agent mode.

Reply:
I refer to it in my blog post as Prometheus in Agent mode.